app/aboot/lk2nd-device.c

```diff
@@ -54,6 +54,22 @@ static void parse_boot_args(void)
 		if ((val = strpresuf(arg, "androidboot.bootloader="))) {
 			lk2nd_dev.bootloader = strdup(val);
 		}
+		if ((val = strpresuf(arg, "androidboot.serialno="))) {
+			lk2nd_dev.barcode = strdup(val);
+		}
+		if ((val = strpresuf(arg, "androidboot.carrier="))) {
+			lk2nd_dev.carrier = strdup(val);
+		}
+		if ((val = strpresuf(arg, "androidboot.device="))) {
+			lk2nd_dev.device = strdup(val);
+		}
+		if ((val = strpresuf(arg, "androidboot.radio="))) {
+			lk2nd_dev.radio = atoui(val);
+			lk2nd_dev.radio_str = "";
+			if (lk2nd_dev.radio == 0) {
+				lk2nd_dev.radio_str = strdup(val);
+			}
```

Comment:
The integer stuff should be handled in `lk2nd-motorola.c`, here you should just set it to `lk2nd_dev.radio` as-is (as string).